---
title: "Figures projet Myco Phospho"
author: "Marie Locard-Paulet"
date: '`r date()`'
output: html_document
---

```{r packages, echo = F, message=FALSE}
require(ggplot2)
require(reshape2)
require(knitr)
require(gplots)


knitr::opts_chunk$set(echo = F)
source("RScripts/Functions/pdfMLP.R")
```

Data from the analysis described in the document `DataInspection_20171205`.

```{r input}
tab <- read.table("OutputTables/NormIntProt_20180110.txt", sep = "\t", header = T)
```

# Comparison of the PknB forms in the samples


```{r}
PknB <- tab[grepl("GN=pknB", as.character(tab$Fasta.headers)),grepl("LFQ", names(tab)) | names(tab)=="Fasta.headers"]

gtab <- melt(PknB)
gtab$Condition <- substr(as.character(gtab$variable), 15, 16)
gtab$Gene <- sapply(as.character(gtab$Fasta.headers), function(x) {
  strsplit(x, "GN=", fixed = T)[[1]][2]
})
gtab$Gene <- sapply(gtab$Gene, function(x) {
  strsplit(x, " ", fixed = T)[[1]][1]
})

gtab$iBAQ <- gtab$value
gtab$iBAQ[gtab$Gene=="pknB"] <- gtab$iBAQ[gtab$Gene=="pknB"]/56
gtab$iBAQ[gtab$Gene=="pknBtub"] <- gtab$iBAQ[gtab$Gene=="pknBtub"]/59
gtab$iBAQ[gtab$Gene=="pknBtubm"] <- gtab$iBAQ[gtab$Gene=="pknBtubm"]/4
gtab <- gtab[gtab$Gene!= "pknBtubm",]

ggplot(gtab, aes(y = log10(value), x = Condition, col = Gene)) + geom_point() + ggtitle("LFQ")
ggplot(gtab, aes(y = log10(iBAQ), x = Condition, col = Gene)) + geom_point() + ggtitle("iBAQ")
```

There would be approximatly 10 times more smeg. than tub.

# Analysis of iBAQ values for proteins of interest

```{r}
tab <- read.table("RAW/Proteome_MQ151217_IBAQ/proteinGroups_151217_IBAQ.txt", sep = "\t", header = T)
iBAQ <- tab[,grepl("iBAQ", names(tab)) | names(tab)=="Fasta.headers"]
iBAQ <- iBAQ[,names(iBAQ) != "iBAQ"]

matmean <- matrix(ncol = 3, nrow = nrow(tab))
iBAQ[iBAQ==0] <- NA
for (i in 1:3) {
  el <- c("L", "P", "K")[i]
  temp <- iBAQ[,grepl(paste0(".", el, "[123]"), names(iBAQ))]
  temp <- apply(temp, 2, "as.numeric")
  l <- list(rowMeans(temp[,1:2], na.rm=T), rowMeans(temp[,3:4], na.rm=T), rowMeans(temp[,5:6], na.rm=T))
  matmean[,i] <- sapply(1:length(l[[1]]), function(x) {
    mean(c(l[[1]][x], l[[2]][x], l[[3]][x]), na.rm = T)
  })
}

colnames(matmean) <-  c("L", "P", "K")
row.names(matmean) <- tab$Fasta.headers

# Histogram:

gtab <- as.data.frame(matmean)
vec <- gtab$L[grepl("mmpL3", row.names(gtab))| grepl("pks13", row.names(gtab))]
ggplot(gtab, aes(x = log10(L))) + geom_histogram(fill = "grey30", alpha = 0.6, col = "grey30") + geom_vline(xintercept = log10(vec), col = "red", size=1.2) + theme_light() + labs(title = "Log10 of iBAQs in the control", subtitle = "highlight mmpL3 and pks13")

```

I retrieve the proteins of interest in the table:

```{r}
prot <- read.table("RAW/ProteinsOfInterest.txt", sep = "\t", header = T, stringsAsFactors = F)
prot <- prot[prot$keep,]
temp <- sapply(as.character(prot$Entry.name), function(x) {
  grepl(paste0("GN=", x), iBAQ$Fasta.headers, ignore.case = T)
})
vec <- vector(length = nrow(prot))
for (i in 1:nrow(prot)) {
  val <- as.character(iBAQ$Fasta.headers[grepl(paste0("GN=", prot$Entry.name[i]), as.character(iBAQ$Fasta.headers), ignore.case = T)])
  if (length(val)==1) {
    vec[i] <- val
  } else if (length(val) != 0) {
    vec[i] <- paste(val, collapse = "||")
  }
}
prot$Fasta.headers <- vec

```

I use the "MSMEG" names to retrive the unmached genes.

```{r}
temp <- sapply(as.character(prot$`mc155`[prot$Fasta.headers=="FALSE"]), function(x) {
  grepl(x, iBAQ$Fasta.headers, ignore.case = T)
})
vec <- vector(length = length(prot$`mc155`[prot$Fasta.headers=="FALSE"]))
for (i in 1:length(prot$`mc155`[prot$Fasta.headers=="FALSE"])) {
  val <- as.character(iBAQ$Fasta.headers[grepl(prot$`mc155`[prot$Fasta.headers=="FALSE"][i], as.character(iBAQ$Fasta.headers), ignore.case = T)])
  if (length(val)==1) {
    vec[i] <- val
  } else if (length(val) != 0) {
    vec[i] <- paste(val, collapse = "||")
  }
}
prot$Fasta.headers[prot$Fasta.headers=="FALSE"] <- vec
write.table(prot, "OutputTables/ProteinsOfInterest2.txt", sep = "\t", row.names = F)
```


********************************************************************************

```{r}
Sys.info()
```