---
title: "Comparison of smeg. proteomes"
author: "Marie Locard-Paulet"
date: '`r date()`'
output: html_document
---

```{r}
prot <- read.table("uniprotsmeg_20180116/uniprotsmeg_20180116_Simple.tab", sep = "\t", header = T, quote = "")

require(knitr)
kable(data.frame(table(prot$Proteomes)))
```

40 IDs are not attributed to any proteome. Why?
There is no data in the Tuberculist column.

# I merge the "proteome" information to the phospho results:

```{r, warning=F}
load("RData/Stat.RData")

tab <- tab[!is.na(tab$Proteins),]

vectab <- tab$Proteins
ltab <- sapply(vectab, strsplit, ";")

vecprot <- vector(length = length(ltab))
for (i in seq_along(ltab)) {
  val <- vectab[[i]]
  val <- sapply(val, strsplit, ";")[[1]]
  if (!is.na(val)) {
    if (length(val)>1) {
      vec <- val
      vec <- as.character(prot$Proteomes[match(vec, prot$Entry)])
      vecprot[i] <- paste(sort(unique(vec), decreasing = F), collapse = ", ")
    } else {
      val <- as.character(prot$Proteomes[prot$Entry==val])
    if (length(val)>0){
      vecprot[i] <- val
      }
    } 
  }
}
```

The "FALSE" are the "Pknbtub" sequences that we added and a "Biognosys7".

```{r}
tab$Proteome <- vecprot
tab$Proteome <- gsub("me, UP", "me; UP", tab$Proteome)
tab$Proteome <- gsub(", UP", "UP", tab$Proteome)
kable(data.frame(table(tab$Proteome)))

tab$GeneID[tab$Proteome=="UP000000757: Chromosome"]
tab$GeneID[tab$Proteome=="UP000006158: Chromosome"]
int <- intersect(tab$GeneID[tab$Proteome=="UP000000757: Chromosome"], tab$GeneID[tab$Proteome=="UP000006158: Chromosome"])
```

the genes `r int` are differently annotated in the two proteomes. Maybe others are too... But it is harder to tell because the gene names do not match.

The proteome `UP000006158` is the proteome that contains the Nter sequence from Pks13 that we see phosphorylated. 

# Table re-annotation:

To simplify the table, I keep only one gene name if possible. I decide to first keep the gene name. If this is not available, I keep the"MSMEG" ID, if there is none I keep the "MSMEI".


I manually check that there is no ID ambiguity in the protein names attribution.

```{r}
vec <- tab$GeneID
lvec <- sapply(vec, strsplit, ";")
lvec1 <- lapply(lvec, function(x) {
  x[!grepl("MSME", x)]
})
lvec2 <- lapply(lvec, function(x) {
  x[grepl("MSMEG", x)]
})
lvec3 <- lapply(lvec, function(x) {
  x[grepl("MSMEI", x)]
})
# Matching protein position:
pos <- tab$Positions.within.proteins
lpos <- sapply(pos, strsplit, ";")

lpos1 <- lapply(1:length(lvec), function(x) {
  lpos[[x]][!grepl("MSME", lvec[[x]])]
})
lpos2 <- lapply(1:length(lvec), function(x) {
  lpos[[x]][grepl("MSMEG", lvec[[x]])]
})
lpos3 <- lapply(1:length(lvec), function(x) {
  lpos[[x]][grepl("MSMEI", lvec[[x]])]
})


for (i in 1:length(lvec)) {
  val1 <- lvec1[[i]]
  val2 <- lvec2[[i]]
  val3 <- lvec3[[i]]
  if (length(val1)>0) {
    vec[i] <- val1
    postemp <- as.numeric(lpos1[[i]])
    if (length(unique(postemp))==1) {
      pos[i] <- unique(postemp)
    } else {
      pos[i] <- paste(unique(postemp), collapse="|")
    }
  } else if (length(val2)>0) {
    vec[i] <- val2
    pos[i] <- lpos2[[i]]
  } else if (length(val3)>0) {
    vec[i] <- val3
    pos[i] <- lpos3[[i]]
  } else {
    vec[i] <- NA
    pos[i] <- NA
  }
}

tab$GeneID2 <- vec
tab$Position2 <- pos

tab$PhosID2 <- paste0(tab$GeneID2, "-", tab$Amino.acid, tab$Position2)

write.table(tab, "OutputTables/FinalTable_20180117.txt", sep = "\t",  row.names = F)
```

**THis generates errors in phosphorylation localisation.**